src/main/java/com/conveyal/datatools/manager/models/EC2Info.java

```diff
@@ -53,4 +57,36 @@ public boolean hasSeparateGraphBuildConfig() {
             buildAmiId != null && !buildAmiId.equals(amiId)
         );
     }
+
+    /**
+     * Returns the appropriate ami ID to use when creating a new ec2 instance during a deploy job.
+     *
+     * @param graphAlreadyBuilt whether or not a graph has already been built. If false, this means a build ami should
+     *                          be used if available.
+     */
+    public String getAmiId(boolean graphAlreadyBuilt) {
+        if (!graphAlreadyBuilt && buildAmiId != null) {
+            return buildAmiId;
+        } else if (amiId != null) {
+            return amiId;
+        } else {
+            return DataManager.getConfigPropertyAsText(AMI_CONFIG_PATH);
```

Comment:
Maybe this final else clause should check whether the config value is missing and log an error if so?

Reply:
There is already a check for null in the DeployJob.